x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/56_search_application_search_with_apikey.yml

```diff
@@ -0,0 +1,299 @@
+                  }
+                ],
+                "restriction": {
+                  "workflows": ["search_application_query"]
+                }
+              }
+            }
+          }
+
+  - match: { name: "search-application-api-key" }
+  - set: { encoded: api_key_encoded  }
+  - set: { id: api_key_id }
+
+# Query Search Application with default parameters:
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application-1
+
+  - match: { hits.total.value: 2 }
+  - match: { hits.hits.0._id: "doc1" }
+  - match: { hits.hits.1._id: "doc2" }
+
+
+# Query Search Application overriding part of the parameters:
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_name: field2
+
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc1" }
+
+# Query Search Application overriding all parameters:
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_name: field3
+            field_value: value3
+
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc2" }
+
+# Query Search Application with list of parameters:
+  - do:
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application-with-list
+        body:
+          params:
+            query_string: value3
+            text_fields:
+              - name: field1
+                boost: 1
+              - name: field2
+                boost: 2
+              - name: field3
+                boost: 3
+
+
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._id: "doc2" }
+
+# Query Search Application with invalid parameter validation:
+  - do:
+      catch: "bad_request"
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_name: field3
+            field_value: 35
+
+# Query Search Application without required parameter:
+  - do:
+      catch: "bad_request"
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: test-search-application
+        body:
+          params:
+            field_value: test
+
+# Query Search Application - not found:
+  - do:
+      catch: forbidden
+      headers:
+        Authorization: ApiKey ${api_key_encoded}
+      search_application.search:
+        name: nonexisting-test-search-application
+        body:
+          params:
+            field_name: field3
+            field_value: value3
+
+# Get Analytics Collection should be rejected due to a workflow restriction
```

Comment:
We could use another two rejection tests to show: 
1. The API key cannot directly read from any indices
2. The API key cannot view itself, i.e. `GET _security/api_key?id=THE_API_KEY_ID`, is rejected with role restriction

Reply:
Good suggestion. I will add those tests.

Reply:
Added in https://github.com/elastic/elasticsearch/pull/96881/commits/43ddad048715b39b83420337d0de99689e7e799c